app/src/components/settings/settings_view/index.tsx

```diff
@@ -0,0 +1,275 @@
+  font-size: 16px;
+  line-height: 18.75px;
+  letter-spacing: 0.4px;
+  margin: 0;
+`
+
+const TextLighter = styled.p`
+  color: ${props => props.theme.colors.textColorLighter};
+  font-size: 12px;
+  line-height: 14.06px;
+  margin: 0;
+`
+
+const ButtonRow = styled.div`
+  display: flex;
+
+  button:first-child {
+    margin-right: 12px;
+  }
+`
+
+const FiltersControls = styled.div<{ disabled?: boolean }>`
+  align-items: center;
+  display: flex;
+  margin-left: auto;
+  margin-right: auto;
+  pointer-events: ${props => (props.disabled ? 'none' : 'initial')};
+
+  @media (min-width: ${props => props.theme.themeBreakPoints.sm}) {
+    margin-left: 0;
+    margin-right: 0;
+    padding-left: 10px;
+  }
+`
+
+const NodeDropdown = styled(Dropdown)`
+  min-width: 170px;
+`
+
+const CustomDropdownItem = styled.div`
+  display: block;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  width: 100%;
+  font-size: ${props => props.theme.fonts.defaultSize};
+  line-height: ${props => props.theme.fonts.defaultLineHeight};
+  display: flex;
+  align-items: center;
+  letter-spacing: 0.4px;
+  color: ${props => props.theme.colors.textColorDark};
+
+  img {
+    margin-right: 10px;
+  }
+`
+
+const StatusBage = styled.div<{ status: boolean }>`
+  width: 6px;
+  height: 6px;
+  margin-right: 8px;
+  border-radius: 3px;
+  background-color: ${props => (props.status ? props.theme.message.colors.ok : props.theme.message.colors.error)};
+`
+
+const Input = styled.input`
+  margin-top: 20px;
+
+  ${TextfieldCSS};
+`
+
+const SettingsViewContainer = () => {
+  const history = useHistory()
+  const context = useConnectedWeb3Context()
+
+  const [current, setCurrent] = useState(0)
+  const [url, setUrl] = useState<string>('')
+  const [isValidUrl, setIsValidUrl] = useState<boolean>(false)
+  const [onlineStatus, setOnlineStatus] = useState<boolean>(false)
+  const [inputPlaceholder, setInputPlaceholder] = useState<string>('')
+
+  const urlObject = getChainSpecificAlternativeUrls(context.networkId)
+
+  const dropdownItem = urlObject.map((item, index) => {
+    return {
+      title: item.name,
+      image: item.name === 'Infura' ? imgInfura : undefined,
+      onClick: () => {
+        setCurrent(index)
+        setUrl(item.rpcUrl)
+      },
+    }
+  })
+
+  dropdownItem.push({
+    title: 'Custom',
+    image: undefined,
+    onClick: () => {
+      setCurrent(dropdownItem.length - 1)
+    },
+  })
+
+  const filterItems = dropdownItem.map(item => {
+    // console.log(index)
+    return {
+      content: (
+        <CustomDropdownItem onClick={item.onClick}>
+          {item.image && <img alt="node" src={item.image} />}
+          {item.title}
+        </CustomDropdownItem>
+      ),
+      secondaryText: '',
+      onClick: item.onClick,
+    }
+  })
+  const checkRpcStatus = async () => {
+    try {
+      await axios.post(url, {
+        id: +new Date(),
+        jsonrpc: '2.0',
+        method: 'eth_getBlockByNumber',
```

Comment:
looks like more providers might support `eth_blockNumber` over `eth_getBlockByNumber`. Check out my comment below about the blockscout endpoint, it seems to only support `eth_blockNumber`

Reply:
Ohh great will change that! Wasn't aware of it...